File: BlockBattleBot/Game.cs
using System.Collections.Generic;

namespace BlockBattleBot
{
    public class Game
    {
        public Settings Settings { get; set; }

        public IDictionary<string, Player> Players { get; set; }

        public string PlayerName { get; set; }

        public Round Round { get; set; }

        public Game()
        {
            Settings = new Settings();
            Players = new Dictionary<string, Player>();
            Round = new Round();
            PlayerName = "";
        }

        public Game(Settings settings, IDictionary<string, Player> players, string playerName, Round round)
        {
            Settings = settings;
            Players = players;
            PlayerName = playerName;
            Round = round;
        }
    }
}

File: src/Game.ts
import { Round } from "./Round";
import { Player } from "./Player";
import { Settings } from "./Settings";

interface Players {
	[name: string]: Player
}

export class Game {
	public settings: Settings;
	public playerName: string;
	private _players: Players;
	public round: Round;
	
	constructor() {
		this.settings = new Settings();
		this.round = new Round();
		this._players = {};
	}
	
	public get players(): Players {
		return this._players;
	}
	
	public addPlayer(name: string) {
		this._players[name] = new Player();
	}
}